src/Microsoft.VisualStudio.SDK.Analyzers.Tests/VSSDK004ProvideAutoLoadAttributeAnalyzerTests.cs

```diff
@@ -0,0 +1,151 @@
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+
+[ProvideAutoLoad(""{F184B08F-C81C-45F6-A57F-5ABD9991F28F}"", PackageAutoLoadFlags.None)]
+class Test : AsyncPackage {
+}
+";
+        this.expect.Locations = new[] { new DiagnosticResultLocation("Test0.cs", 4, 2, 4, 86) };
+        this.VerifyCSharpDiagnostic(test, this.expect);
+    }
+
+    [Fact]
+    public void MultipleProvideAutoLoadProducesMultipleDiagnostics()
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+
+[ProvideAutoLoad(""{A184B08F-C81C-45F6-A57F-5ABD9991F28F}"", PackageAutoLoadFlags.None)]
+[ProvideAutoLoad(""{F184B08F-C81C-45F6-A57F-5ABD9991F28F}"", PackageAutoLoadFlags.None)]
+class Test : AsyncPackage {
+}
+";
+        DiagnosticResult[] expected = new[]
+        {
+            new DiagnosticResult()
+            {
+                Id = VSSDK004ProvideAutoLoadAttributeAnalyzer.Id,
+                SkipVerifyMessage = true,
+                Severity = DiagnosticSeverity.Warning,
+                Locations = new[] { new DiagnosticResultLocation("Test0.cs", 4, 2, 4, 86) },
+            },
+            new DiagnosticResult()
+            {
+                Id = VSSDK004ProvideAutoLoadAttributeAnalyzer.Id,
+                SkipVerifyMessage = true,
+                Severity = DiagnosticSeverity.Warning,
+                Locations = new[] { new DiagnosticResultLocation("Test0.cs", 5, 2, 5, 86) },
+            },
+        };
+
+        this.VerifyCSharpDiagnostic(test, expected);
+    }
+
+    [Fact]
+    public void ProvideAutoLoadWithNamedFlagsButNoBackgroundLoadProducesDiagnostics()
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+
+[ProvideAutoLoad(""{F184B08F-C81C-45F6-A57F-5ABD9991F28F}"", flags: PackageAutoLoadFlags.None)]
+class Test : AsyncPackage {
+}
+";
+        this.expect.Locations = new[] { new DiagnosticResultLocation("Test0.cs", 4, 2, 4, 93) };
+        this.VerifyCSharpDiagnostic(test, this.expect);
+    }
+
+    [Fact]
+    public void ProvideAutoLoadWithSkipFlagProducesNoDiagnostics()
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+
+[ProvideAutoLoad(""{F184B08F-C81C-45F6-A57F-5ABD9991F28F}"", PackageAutoLoadFlags.SkipWhenUIContextRulesActive)]
+class Test : AsyncPackage {
+}
+";
+        this.VerifyCSharpDiagnostic(test);
+    }
+
+    [Fact]
+    public void ProvideAutoLoadWithBackgroundLoadFlagProducesNoDiagnostics()
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+
+[ProvideAutoLoad(""{F184B08F-C81C-45F6-A57F-5ABD9991F28F}"", PackageAutoLoadFlags.BackgroundLoad)]
+class Test : AsyncPackage {
+}
+";
+        this.VerifyCSharpDiagnostic(test);
+    }
+
+    [Fact]
+    public void ProvideAutoLoadOnPackageWithBackgroundLoadFlagProducesNoDiagnostics()
+    {
+        var test = @"
+using Microsoft.VisualStudio.Shell;
+
+[ProvideAutoLoad(""{F184B08F-C81C-45F6-A57F-5ABD9991F28F}"", PackageAutoLoadFlags.BackgroundLoad)]
```

Comment:
All of your tests seem to use the attribute constructor that takes `PackageAutoLoadFlags` as a second parameter. Can you add tests where you use the 1-parameter constructor and pass in the flags as a named argument? I think you'll find that tests fails, since your analyzer currently doesn't check the `AttributeSyntax.NamedArguments` collection.

Reply:
In ProvideAutoLoad, the properties are getters only so you are not able to use named parameters.

Reply:
Oh! Ok.